Lesson_2_3/BankAccount.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.IO;
+using System.Text.Json;
+using Newtonsoft.Json;
+
+namespace Lesson_2_3 {
+
+
+    public class BankAccount
+    {
+        //public static Dictionary<long, BankAccount> Accounts = new Dictionary<long, BankAccount>();
+        private int _Id;
+        private decimal _Balance;
+        private string _Type, _User_Name;
+
+
+        public BankAccount(int Balance, string Type, string User_Name)
+        {
+            _Id = Generate_Id.Generate_id();
+            _Balance = Balance;
+            _Type = Type;
+            _User_Name = User_Name;
+           
+
+        }
+        
+        public bool WithDraw(decimal sum)
+        {
+            _Balance -= sum;
+            return _Balance >= sum;
+
+        }
+        public bool PutOn(decimal sum)
+        {
+            _Balance += sum;
+            return true;
+        }
+
+        public bool TarnsferMoney(BankAccount Sourse, decimal Ammount)
+        {
+            if (Sourse._Balance >= Ammount)
+            {
+                Sourse._Balance -= Ammount;
+                _Balance += Ammount;
+                return true;
+            }
+            else
+                return false;
+        }
+
+
+        public static bool operator ==(BankAccount a, BankAccount b)
+        {
+            return a._Balance == b._Balance && a._Type == b._Type && a._User_Name == b._User_Name;
```

Comment:
Если а, или б будет равно null, то ляжет всё приложение.
```suggestion
            return Equals(a, b);
```